src/decoder/external_decoder.ml

```diff
@@ -150,11 +149,12 @@ let external_input_oblivious process filename prebuf =
     try Process_handler.kill process with Process_handler.Finished -> ()
   in
   let input = { Decoder.read; tell = None; length = None; lseek = None } in
-  let kind = Frame.{ audio = Any; video = Zero; midi = Zero } in
-  let gen =
-    Generator.create ~log_overfull:false ~log:(log#info "%s") ~kind `Audio
+  (* TODO: is this really what we want for audio channels? *)
+  let ctype =
+    Frame.{ audio = Lazy.force Frame.audio_channels; video = 0; midi = 0 }
```

Comment:
Same thing here, we should be able to trust our implicit channel conversion now. I haven't looked at the details of this part of the code, though.